src/qt/addresstablemodel.cpp

```diff
@@ -82,14 +82,14 @@ class AddressTablePriv
             LOCK(wallet->cs_wallet);
             for (const std::pair<CTxDestination, CAddressBookData>& item : wallet->mapAddressBook)
             {
-                const CBitcoinAddress& address = item.first;
```

Comment:
It was creating a temporary of type CBitcoinAddress by converting the CTxDestination initializer, using the implicit `CBitcoinAddress(const CTxDestination &dest)` constructor. That temporary's lifetime was then extended by taking a reference to it, named `address`.